### ISSUE TITLE
Bridge GitHub Actions and Jenkins

### ISSUE THREAD

**Issue by cbtkc** [department-of-veterans-affairs/va.gov-team#23096]
AC:
- [ ] Github action exists that triggers jenkins deployment for non-prod environments (For master only)
- [x] Remove CI jobs in Jenkins

**Comment by ericboehs** [comment]
Just work on the first AC here until we are able to get dedicated self-hosted runners that aren't ephemeral.

**Comment by jeff2d2** [comment]
Pull these into their own ticket:
- [ ] Update docs

**Comment by cbtkc** [comment]
PR has been created to add the dev and staging deployment jobs to the allowed_api_endpoints array, allowing them to be triggered by GHA https://github.com/department-of-veterans-affairs/devops/pull/9102

**Comment by cbtkc** [comment]
Additional devops PR has been created to add the AMI build job to the allowed endpoints list, which will allow this job to be triggered via a Github action. This build will need to be triggered prior to the dev & staging deployment jobs being triggered, as they require an AMI to be built prior to deployment.

**Comment by cbtkc** [comment]
Added another PR to DevOps repo to allow us to check the build status of the vets-api AMI build job. This will need to be checked and verified as successful prior to kicking off the deployment jobs.

https://github.com/department-of-veterans-affairs/devops/pull/9218

**Comment by cbtkc** [comment]
Waiting on @rileyanderson to finish his work on getting docker working within GHA. After that is done, I can remove the remaining tests.

**Comment by ericboehs** [comment]
I believe I meant to comment this here instead of on #24536:

>We've moved away as much as makes sense for now as we're moving to K8s/argo in ~6 months. Everything is being ran in GHA except when we deploy (Jenkins runs our rspec suite on deploy, for now).
